### DISCUSSION TITLE
v1.5.0-rc.1

### DISCUSSION THREAD

**Discussion by ruslandoga** [plausible/analytics#2460] [Anything Else]
Admin user registration is now done through the UI instead of configuration variables. At first launch, users are presented with a registration page which will create the admin account. Admin account related configuration variables are ignored going forward. See below for details.

Admin user registration is required from 1.5.0 onwards. We are deprecating the `DISABLE_AUTH` option which allowed Plausible to be used without any user account at all.

Users with existing admin accounts are not affected, but we recommend removing the unused configuration variables. A benefit of moving admin account management to the UI is that you can now change the admin email account through the UI. Previously, this was done by manual database changes and configuration change which required restarting the service.

For `v1.5.0-rc.0` changes and upgrade steps please see [the previous release candidate discussion.](https://github.com/plausible/analytics/discussions/2324)

## Upgrading Plausible Analytics to v1.5.0-rc.1

In your `docker-compose.yml` update the image used for `plausible` to tag `v1.5.0-rc.1`:

```diff
plausible:
- image: plausible/analytics:1.5.0-rc.0
+ image: plausible/analytics:v1.5.0-rc.1
````

#### Restart the container

```
$ docker compose stop plausible
$ docker compose rm plausible
$ docker compose up -d
```

This will boot up the new version of the app.

### Removing deprecated commands and environment variables (Optional, but recommended)

`ADMIN_USER_EMAIL`, `ADMIN_USER_NAME`, `ADMIN_USER_PWD`, `DISABLE_AUTH` environment variables are no longer supported and are ignored if set.

Please remove them from `plausible-conf.env`:

```diff
- ADMIN_USER_EMAIL=replace-me
- ADMIN_USER_NAME=replace-me
- ADMIN_USER_PWD=replace-me
- DISABLE_AUTH=true
  BASE_URL=replace-me
  SECRET_KEY_BASE=replace-me
```

[`/entrypoint.sh db init-admin`](https://github.com/plausible/hosting/blob/a6fe8d13df3566f7bb3625a1a524f5498d56693d/docker-compose.yml#L31) command that was responsible for creating the admin user [is also deprecated.](https://github.com/plausible/analytics/blob/e522a2d7c182f5a0c7b2d95bf3ea248567a529e0/rel/overlays/init-admin.sh#L4-L5)

Please remove it from `docker-compose.yml`:

```diff
plausible:
  image: plausible/analytics:v1.5.0-rc.1
- command: sh -c "sleep 10 && /entrypoint.sh db createdb && /entrypoint.sh db migrate && /entrypoint.sh db init-admin && /entrypoint.sh run"
+ command: sh -c "sleep 10 && /entrypoint.sh db createdb && /entrypoint.sh db migrate && /entrypoint.sh run"
```

## Changelog

Following [changes](https://github.com/plausible/analytics/compare/v1.5.0-rc.0...v1.5.0-rc.1) have been made since `v1.5.0-rc.0`:

### Added

- Allow self-hosters to register an account on first launch

### Deprecated

- `DISABLE_AUTH` is no longer supported

<hr /><em>This discussion was created from the release <a href='https://github.com/plausible/analytics/releases/tag/v1.5.0-rc.1'>v1.5.0-rc.1</a>.</em>

**Comment by timscha** [comment]
I got this error when following the update instructions:

Error response from daemon: manifest for plausible/analytics:1.5.0-rc.1 not found: manifest unknown: manifest unknown

**Reply by ruslandoga** [reply]
The image tag is `plausible/analytics:v1.5.0-rc.1`, please note the `v`.

The missing `v` in `1.5.0-rc.0` was a mistake on our part: https://github.com/plausible/analytics/discussions/2324#discussioncomment-3867370

**Comment by timscha** [comment]
Upsi, I missed the v. Thanks @ruslandoga  :)

**Comment by joshavant** [comment]
@ruslandoga Thanks for the great work on this.

Is the Geonames CSV necessary to get city-level geolocation working?

Does the official Plausible Docker download the Geonames CSV, or is there a Docker Hub source for this?

**Reply by ruslandoga** [reply]
@joshavant 👋

Yeah, geonames are necessary as they provide the mapping of geoname id -> human-readable place name. [Here's](https://github.com/plausible/analytics/discussions/2324#discussioncomment-3857114) a guide to setup city-level geolocation.

**Comment by mattvb91** [comment]
How do I create a user now without [/entrypoint.sh db init-admin](https://github.com/plausible/hosting/blob/a6fe8d13df3566f7bb3625a1a524f5498d56693d/docker-compose.yml#L31)? I boot my full stack in github actions to make sure analytics etc.. is working throughout my apps with various events but now I need to manually register via web ui? How do you create it via script now?

**Reply by ruslandoga** [reply]
👋 @mattvb91

You might be able to mount a custom [`priv/repo/seeds.exs` script](https://github.com/plausible/analytics/blob/master/priv/repo/seeds.exs) which'd create a user and some sites and then run it with [`bin/plausible eval "Plausible.Release.seed()"`](https://github.com/plausible/analytics/blob/138e7c06d66c730414b82e9ff8479430bec9d22d/lib/plausible_release.ex#L27-L33) (I'll make a demo github ci action with this approach today)

**Reply by ruslandoga** [reply]
Demo: https://github.com/ruslandoga/plausible-seeds-ci

**Reply by mattvb91** [reply]
Thank you! Apreciated :+1:

**Comment by ruslandoga** [comment]
[`v1.5.1`](https://github.com/plausible/analytics/discussions/2507) has been released.

**Comment by ErikHumphrey** [comment]
Any recommended replacement for `DISABLE_AUTH` for those using an OAuth2 proxy, Keycloak, etc. in front of Plausible? I.e., to automatically sign in as the admin user if the app is otherwise secured.

**Reply by ruslandoga** [reply]
👋 @ErikHumphrey 

I don't think there is any replacement for `DISABLE_AUTH` right now.

**Comment by ruslandoga** [comment]
👋

https://github.com/plausible/analytics/discussions/2896 `v2.0.0-rc.0` is now available.
